acquia_lift.admin.wizard.inc

```diff
@@ -444,6 +561,53 @@ function acquia_lift_personalize_campaign_wizard_variations_submit_element_edit(
 }
 
 /**
+ * Submit function for targeting form.
+ */
+function acquia_lift_personalize_campaign_wizard_targeting_submit(&$form, &$form_state, &$agent_data, $agent_instance) {
+  // Save any visitor context if passed.
+  // Add the visitor context configuration.
+  $visitor_context = array();
+  if (isset($form_state['values']['visitor_context'])) {
+    $visitor_context = personalize_admin_convert_visitor_context_form_values($form_state['values']['visitor_context']);
+  }
+  $agent_data->data['visitor_context'] = $visitor_context;
+
+  // Handle targeting audiences.
+  if (empty($form_state['values']['audiences'])) {
+    return;
+  }
+  module_load_include('inc', 'acquia_lift', 'acquia_lift.admin');
+  foreach ($form_state['values']['audiences'] as $rule_id => $rule_values) {
```

Comment:
Can we call them audiences instead of rules? I think it could get confusing otherwise as an audience is made up of rules.